sdk/eventhub/Microsoft.Azure.WebJobs.Extensions.EventHubs/src/Config/EventHubExtensionConfigProvider.cs

```diff
@@ -0,0 +1,136 @@
+            }
+
+            _options.Value.SetExceptionHandler(ExceptionReceivedHandler);
+            _configuration.ConfigurationSection.Bind(_options);
+
+            context
+                .AddConverter<string, EventData>(ConvertString2EventData)
+                .AddConverter<EventData, string>(ConvertEventData2String)
+                .AddConverter<byte[], EventData>(ConvertBytes2EventData)
+                .AddConverter<EventData, byte[]>(ConvertEventData2Bytes)
+                .AddOpenConverter<OpenType.Poco, EventData>(ConvertPocoToEventData);
+
+            // register our trigger binding provider
+            var triggerBindingProvider = new EventHubTriggerAttributeBindingProvider(_config, _nameResolver, _converterManager, _options, _loggerFactory);
+            context.AddBindingRule<EventHubTriggerAttribute>()
+                .BindToTrigger(triggerBindingProvider);
+
+            // register our binding provider
+            context.AddBindingRule<EventHubAttribute>()
+                .BindToCollector(BuildFromAttribute);
+
+            context.AddBindingRule<EventHubAttribute>()
+                .BindToInput(attribute =>
+            {
+                return _options.Value.GetEventHubProducerClient(attribute.EventHubName, attribute.Connection);
+            });
+
+            ExceptionHandler = (e =>
+            {
+                LogExceptionReceivedEvent(e, _loggerFactory);
+            });
+        }
+
+        internal static void LogExceptionReceivedEvent(ExceptionReceivedEventArgs e, ILoggerFactory loggerFactory)
+        {
+            var logger = loggerFactory?.CreateLogger(LogCategories.Executor);
+            string message = $"EventProcessorHost error (Action='{e.Action}', HostName='{e.Hostname}', PartitionId='{e.PartitionId}').";
+
+            Utility.LogException(e.Exception, message, logger);
+        }
+
+        private static LogLevel GetLogLevel(Exception ex)
+        {
+            var ehex = ex as EventHubsException;
+            if (!(ex is OperationCanceledException) && (ehex == null || !ehex.IsTransient))
+            {
+                // any non-transient exceptions or unknown exception types
+                // we want to log as errors
+                return LogLevel.Error;
+            }
+            else
+            {
+                // transient messaging errors we log as info so we have a record
+                // of them, but we don't treat them as actual errors
+                return LogLevel.Information;
+            }
+        }
+
+        private IAsyncCollector<EventData> BuildFromAttribute(EventHubAttribute attribute)
+        {
+            EventHubProducerClient client = _options.Value.GetEventHubProducerClient(attribute.EventHubName, attribute.Connection);
+            return new EventHubAsyncCollector(new EventHubProducerClientImpl(client));
+        }
+
+        private static string ConvertEventData2String(EventData x)
```

Comment:
I'm going to assume that you're porting these over, but.... really?   They're abbreviating "To" for a single character savings?   
![](https://media3.giphy.com/media/3e18NPUVzoxzO/giphy.gif)